NoneType：object has no attribute 'event_hooks'报错

## 环境

- nonebot-bison 版本：0.9.3
- nonebot 版本：2.3.2
- 安装方式：（以下方式的一种或者其他方式）
  1. 通过 nb-cli 安装
- 操作系统：
- Windows10

## 问题

请在这里描述你遇到的问题
后台报错，真实环境运行报错，又建了个虚拟环境重新装了nonebot2还是报错
## 日志
```log
08-07 01:41:33 [INFO] websockets | connection open
08-07 01:41:33 [INFO] nonebot_plugin_saa | refresh bot platform target cache Bot(type='OneBot V11', self_id='1111111111')
08-07 01:42:04 [INFO] nonebot_bison | init rss-https://ngs.pso2-makapo.com/feed/ with set()
08-07 01:42:25 [INFO] nonebot_bison | init bilibili-120186 with set()
08-07 01:42:32 [ERROR] apscheduler | Job "Scheduler.exec_fetch (trigger: interval[0:01:00], next run at: 2024-08-07 01:43:32 CST)" raised an exception
Traceback (most recent call last):
  File "G:\NoneBot2\Nonebot2E\bot.py", line 54, in <module>
    nonebot.run(app="__mp_main__:app")
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot\__init__.py", line 335, in run
    get_driver().run(*args, **kwargs)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run
    uvicorn.run(
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\uvicorn\main.py", line 577, in run
    server.run()
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\uvicorn\server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever
    self._run_once()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
> File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
    retval = await job.func(*job.args, **job.kwargs)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\scheduler\scheduler.py", line 109, in exec_fetch
    to_send = await platform_obj.do_fetch_new_post(SubUnit(schedulable.target, send_userinfo_list))
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 108, in do_fetch_new_post
    return await catch_network_error(self.fetch_new_post, sub_unit) or []
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 53, in catch_network_error
    return await func(*args, **kwargs)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 342, in fetch_new_post
    post_list = await self.get_sub_list(sub_unit.sub_target)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\ff14.py", line 27, in get_sub_list
    client = await self.ctx.get_client()
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\utils\context.py", line 53, in get_client
    self._register_to_client(client)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\utils\context.py", line 28, in _register_to_client
    client.event_hooks = hooks
AttributeError: ("'NoneType' object has no attribute 'event_hooks'", [])
08-07 01:43:32 [ERROR] apscheduler | Job "Scheduler.exec_fetch (trigger: interval[0:01:00], next run at: 2024-08-07 01:44:32 CST)" raised an exception
Traceback (most recent call last):
  File "G:\NoneBot2\Nonebot2E\bot.py", line 54, in <module>
    nonebot.run(app="__mp_main__:app")
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot\__init__.py", line 335, in run
    get_driver().run(*args, **kwargs)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run
    uvicorn.run(
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\uvicorn\main.py", line 577, in run
    server.run()
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\uvicorn\server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever
    self._run_once()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
> File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
    retval = await job.func(*job.args, **job.kwargs)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\scheduler\scheduler.py", line 109, in exec_fetch
    to_send = await platform_obj.do_fetch_new_post(SubUnit(schedulable.target, send_userinfo_list))
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 108, in do_fetch_new_post
    return await catch_network_error(self.fetch_new_post, sub_unit) or []
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 53, in catch_network_error
    return await func(*args, **kwargs)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 342, in fetch_new_post
    post_list = await self.get_sub_list(sub_unit.sub_target)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\ff14.py", line 27, in get_sub_list
    client = await self.ctx.get_client()
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\utils\context.py", line 53, in get_client
    self._register_to_client(client)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\utils\context.py", line 28, in _register_to_client
    client.event_hooks = hooks
AttributeError: ("'NoneType' object has no attribute 'event_hooks'", [])
08-07 01:44:32 [ERROR] apscheduler | Job "Scheduler.exec_fetch (trigger: interval[0:01:00], next run at: 2024-08-07 01:45:32 CST)" raised an exception
Traceback (most recent call last):
  File "G:\NoneBot2\Nonebot2E\bot.py", line 54, in <module>
    nonebot.run(app="__mp_main__:app")
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot\__init__.py", line 335, in run
    get_driver().run(*args, **kwargs)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run
    uvicorn.run(
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\uvicorn\main.py", line 577, in run
    server.run()
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\uvicorn\server.py", line 65, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete
    self.run_forever()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
    super().run_forever()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever
    self._run_once()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1909, in _run_once
    handle._run()
  File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
> File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
    retval = await job.func(*job.args, **job.kwargs)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\scheduler\scheduler.py", line 109, in exec_fetch
    to_send = await platform_obj.do_fetch_new_post(SubUnit(schedulable.target, send_userinfo_list))
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 108, in do_fetch_new_post
    return await catch_network_error(self.fetch_new_post, sub_unit) or []
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 53, in catch_network_error
    return await func(*args, **kwargs)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 342, in fetch_new_post
    post_list = await self.get_sub_list(sub_unit.sub_target)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\platform\ff14.py", line 27, in get_sub_list
    client = await self.ctx.get_client()
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\utils\context.py", line 53, in get_client
    self._register_to_client(client)
  File "G:\NoneBot2\Nonebot2E\.venv\lib\site-packages\nonebot_bison\utils\context.py", line 28, in _register_to_client
    client.event_hooks = hooks
AttributeError: ("'NoneType' object has no attribute 'event_hooks'", [])
```

![image](https://github.com/user-attachments/assets/42c99644-45af-4d69-9531-f81e6f63e920)

- [x] 我搜索过了 issue，但是并没有发现过与我类似的问题
- [x] 我确认在日志中去掉了敏感信息

> ## 环境
> * nonebot-bison 版本：0.9.3
> * nonebot 版本：2.3.2
> * 安装方式：（以下方式的一种或者其他方式）
>   
>   1. 通过 nb-cli 安装
> * 操作系统：
> * Windows10
> 
> ## 问题
> 请在这里描述你遇到的问题 后台报错，真实环境运行报错，又建了个虚拟环境重新装了nonebot2还是报错
> 
> ## 日志
> 08-07 01:41:33 [INFO] websockets | connection open 08-07 01:41:33 [INFO] nonebot_plugin_saa | refresh bot platform target cache Bot(type='OneBot V11', self_id='1111111111') 08-07 01:42:04 [INFO] nonebot_bison | init rss-https://ngs.pso2-makapo.com/feed/ with set() 08-07 01:42:25 [INFO] nonebot_bison | init bilibili-120186 with set() 08-07 01:42:32 [ERROR] apscheduler | Job "Scheduler.exec_fetch (trigger: interval[0:01:00], next run at: 2024-08-07 01:43:32 CST)" raised an exception Traceback (most recent call last): File "G:\NoneBot2\Nonebot2E\bot.py", line 54, in nonebot.run(app="**mp_main**:app") File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot__init__.py", line 335, in run get_driver().run(*args, **kwargs) File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run uvicorn.run( File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\main.py", line 577, in run server.run() File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\server.py", line 65, in run return asyncio.run(self.serve(sockets=sockets)) File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run return loop.run_until_complete(main) File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete self.run_forever() File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever super().run_forever() File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever self._run_once() File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1909, in _run_once handle._run() File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run self._context.run(self._callback, *self._args)
> 
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
> > retval = await job.func(*job.args, **job.kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\scheduler\scheduler.py", line 109, in exec_fetch
> > to_send = await platform_obj.do_fetch_new_post(SubUnit(schedulable.target, send_userinfo_list))
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 108, in do_fetch_new_post
> > return await catch_network_error(self.fetch_new_post, sub_unit) or []
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 53, in catch_network_error
> > return await func(*args, **kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 342, in fetch_new_post
> > post_list = await self.get_sub_list(sub_unit.sub_target)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\ff14.py", line 27, in get_sub_list
> > client = await self.ctx.get_client()
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 53, in get_client
> > self._register_to_client(client)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 28, in _register_to_client
> > client.event_hooks = hooks
> > AttributeError: ("'NoneType' object has no attribute 'event_hooks'", [])
> > 08-07 01:43:32 [ERROR] apscheduler | Job "Scheduler.exec_fetch (trigger: interval[0:01:00], next run at: 2024-08-07 01:44:32 CST)" raised an exception
> > Traceback (most recent call last):
> > File "G:\NoneBot2\Nonebot2E\bot.py", line 54, in 
> > nonebot.run(app="**mp_main**:app")
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot__init__.py", line 335, in run
> > get_driver().run(*args, **kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run
> > uvicorn.run(
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\main.py", line 577, in run
> > server.run()
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\server.py", line 65, in run
> > return asyncio.run(self.serve(sockets=sockets))
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
> > return loop.run_until_complete(main)
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete
> > self.run_forever()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
> > super().run_forever()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever
> > self._run_once()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1909, in _run_once
> > handle._run()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
> > self._context.run(self._callback, *self._args)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
> > retval = await job.func(*job.args, **job.kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\scheduler\scheduler.py", line 109, in exec_fetch
> > to_send = await platform_obj.do_fetch_new_post(SubUnit(schedulable.target, send_userinfo_list))
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 108, in do_fetch_new_post
> > return await catch_network_error(self.fetch_new_post, sub_unit) or []
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 53, in catch_network_error
> > return await func(*args, **kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 342, in fetch_new_post
> > post_list = await self.get_sub_list(sub_unit.sub_target)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\ff14.py", line 27, in get_sub_list
> > client = await self.ctx.get_client()
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 53, in get_client
> > self._register_to_client(client)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 28, in _register_to_client
> > client.event_hooks = hooks
> > AttributeError: ("'NoneType' object has no attribute 'event_hooks'", [])
> > 08-07 01:44:32 [ERROR] apscheduler | Job "Scheduler.exec_fetch (trigger: interval[0:01:00], next run at: 2024-08-07 01:45:32 CST)" raised an exception
> > Traceback (most recent call last):
> > File "G:\NoneBot2\Nonebot2E\bot.py", line 54, in 
> > nonebot.run(app="**mp_main**:app")
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot__init__.py", line 335, in run
> > get_driver().run(*args, **kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run
> > uvicorn.run(
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\main.py", line 577, in run
> > server.run()
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\server.py", line 65, in run
> > return asyncio.run(self.serve(sockets=sockets))
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
> > return loop.run_until_complete(main)
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete
> > self.run_forever()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
> > super().run_forever()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever
> > self._run_once()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1909, in _run_once
> > handle._run()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
> > self._context.run(self._callback, *self._args)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
> > retval = await job.func(*job.args, **job.kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\scheduler\scheduler.py", line 109, in exec_fetch
> > to_send = await platform_obj.do_fetch_new_post(SubUnit(schedulable.target, send_userinfo_list))
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 108, in do_fetch_new_post
> > return await catch_network_error(self.fetch_new_post, sub_unit) or []
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 53, in catch_network_error
> > return await func(*args, **kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 342, in fetch_new_post
> > post_list = await self.get_sub_list(sub_unit.sub_target)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\ff14.py", line 27, in get_sub_list
> > client = await self.ctx.get_client()
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 53, in get_client
> > self._register_to_client(client)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 28, in _register_to_client
> > client.event_hooks = hooks
> > AttributeError: ("'NoneType' object has no attribute 'event_hooks'", [])
> 
> ```
> 请在这里粘贴你的日志
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/169012523/355550267-42c99644-45af-4d69-9531-f81e6f63e920.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5NjY5MDQsIm5iZiI6MTcyMjk2NjYwNCwicGF0aCI6Ii8xNjkwMTI1MjMvMzU1NTUwMjY3LTQyYzk5NjQ0LTQ1YWYtNGQ2OS05NTMxLWY4MWU2ZjYzZTkyMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwNlQxNzUwMDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02M2E4ZTY0ZmMzOTFjYTc3OWY4ZWIyOTZlNzRlN2JhZmNjYjc1NmIyMzNkOTMwN2E2ZjgxNzA0YTY0NTU0MmNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Ny753swM5fu9Qq_xtVB5gZ4u27tNvK_mGl7w5Mr2Fb0)
> 
> * [x]  我搜索过了 issue，但是并没有发现过与我类似的问题
> * [x]  我确认在日志中去掉了敏感信息



> ## 环境
> * nonebot-bison 版本：0.9.3
> * nonebot 版本：2.3.2
> * 安装方式：（以下方式的一种或者其他方式）
>   
>   1. 通过 nb-cli 安装
> * 操作系统：
> * Windows10
> 
> ## 问题
> 请在这里描述你遇到的问题 后台报错，真实环境运行报错，又建了个虚拟环境重新装了nonebot2还是报错
> 
> ## 日志
> 08-07 01:41:33 [INFO] websockets | connection open 08-07 01:41:33 [INFO] nonebot_plugin_saa | refresh bot platform target cache Bot(type='OneBot V11', self_id='1111111111') 08-07 01:42:04 [INFO] nonebot_bison | init rss-https://ngs.pso2-makapo.com/feed/ with set() 08-07 01:42:25 [INFO] nonebot_bison | init bilibili-120186 with set() 08-07 01:42:32 [ERROR] apscheduler | Job "Scheduler.exec_fetch (trigger: interval[0:01:00], next run at: 2024-08-07 01:43:32 CST)" raised an exception Traceback (most recent call last): File "G:\NoneBot2\Nonebot2E\bot.py", line 54, in nonebot.run(app="**mp_main**:app") File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot__init__.py", line 335, in run get_driver().run(*args, **kwargs) File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run uvicorn.run( File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\main.py", line 577, in run server.run() File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\server.py", line 65, in run return asyncio.run(self.serve(sockets=sockets)) File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run return loop.run_until_complete(main) File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete self.run_forever() File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever super().run_forever() File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever self._run_once() File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1909, in _run_once handle._run() File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run self._context.run(self._callback, *self._args)
> 
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
> > retval = await job.func(*job.args, **job.kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\scheduler\scheduler.py", line 109, in exec_fetch
> > to_send = await platform_obj.do_fetch_new_post(SubUnit(schedulable.target, send_userinfo_list))
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 108, in do_fetch_new_post
> > return await catch_network_error(self.fetch_new_post, sub_unit) or []
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 53, in catch_network_error
> > return await func(*args, **kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 342, in fetch_new_post
> > post_list = await self.get_sub_list(sub_unit.sub_target)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\ff14.py", line 27, in get_sub_list
> > client = await self.ctx.get_client()
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 53, in get_client
> > self._register_to_client(client)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 28, in _register_to_client
> > client.event_hooks = hooks
> > AttributeError: ("'NoneType' object has no attribute 'event_hooks'", [])
> > 08-07 01:43:32 [ERROR] apscheduler | Job "Scheduler.exec_fetch (trigger: interval[0:01:00], next run at: 2024-08-07 01:44:32 CST)" raised an exception
> > Traceback (most recent call last):
> > File "G:\NoneBot2\Nonebot2E\bot.py", line 54, in 
> > nonebot.run(app="**mp_main**:app")
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot__init__.py", line 335, in run
> > get_driver().run(*args, **kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run
> > uvicorn.run(
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\main.py", line 577, in run
> > server.run()
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\server.py", line 65, in run
> > return asyncio.run(self.serve(sockets=sockets))
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
> > return loop.run_until_complete(main)
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete
> > self.run_forever()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
> > super().run_forever()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever
> > self._run_once()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1909, in _run_once
> > handle._run()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
> > self._context.run(self._callback, *self._args)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
> > retval = await job.func(*job.args, **job.kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\scheduler\scheduler.py", line 109, in exec_fetch
> > to_send = await platform_obj.do_fetch_new_post(SubUnit(schedulable.target, send_userinfo_list))
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 108, in do_fetch_new_post
> > return await catch_network_error(self.fetch_new_post, sub_unit) or []
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 53, in catch_network_error
> > return await func(*args, **kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 342, in fetch_new_post
> > post_list = await self.get_sub_list(sub_unit.sub_target)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\ff14.py", line 27, in get_sub_list
> > client = await self.ctx.get_client()
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 53, in get_client
> > self._register_to_client(client)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 28, in _register_to_client
> > client.event_hooks = hooks
> > AttributeError: ("'NoneType' object has no attribute 'event_hooks'", [])
> > 08-07 01:44:32 [ERROR] apscheduler | Job "Scheduler.exec_fetch (trigger: interval[0:01:00], next run at: 2024-08-07 01:45:32 CST)" raised an exception
> > Traceback (most recent call last):
> > File "G:\NoneBot2\Nonebot2E\bot.py", line 54, in 
> > nonebot.run(app="**mp_main**:app")
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot__init__.py", line 335, in run
> > get_driver().run(*args, **kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot\drivers\fastapi.py", line 186, in run
> > uvicorn.run(
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\main.py", line 577, in run
> > server.run()
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\uvicorn\server.py", line 65, in run
> > return asyncio.run(self.serve(sockets=sockets))
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\runners.py", line 44, in run
> > return loop.run_until_complete(main)
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 636, in run_until_complete
> > self.run_forever()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\windows_events.py", line 321, in run_forever
> > super().run_forever()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 603, in run_forever
> > self._run_once()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\base_events.py", line 1909, in _run_once
> > handle._run()
> > File "C:\Users\Leo\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 80, in _run
> > self._context.run(self._callback, *self._args)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\apscheduler\executors\base_py3.py", line 30, in run_coroutine_job
> > retval = await job.func(*job.args, **job.kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\scheduler\scheduler.py", line 109, in exec_fetch
> > to_send = await platform_obj.do_fetch_new_post(SubUnit(schedulable.target, send_userinfo_list))
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 108, in do_fetch_new_post
> > return await catch_network_error(self.fetch_new_post, sub_unit) or []
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 53, in catch_network_error
> > return await func(*args, **kwargs)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\platform.py", line 342, in fetch_new_post
> > post_list = await self.get_sub_list(sub_unit.sub_target)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\platform\ff14.py", line 27, in get_sub_list
> > client = await self.ctx.get_client()
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 53, in get_client
> > self._register_to_client(client)
> > File "G:\NoneBot2\Nonebot2E.venv\lib\site-packages\nonebot_bison\utils\context.py", line 28, in _register_to_client
> > client.event_hooks = hooks
> > AttributeError: ("'NoneType' object has no attribute 'event_hooks'", [])
> 
> ```
> 请在这里粘贴你的日志
> ```
> 
> ![image](https://private-user-images.githubusercontent.com/169012523/355550267-42c99644-45af-4d69-9531-f81e6f63e920.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5NjY5MDQsIm5iZiI6MTcyMjk2NjYwNCwicGF0aCI6Ii8xNjkwMTI1MjMvMzU1NTUwMjY3LTQyYzk5NjQ0LTQ1YWYtNGQ2OS05NTMxLWY4MWU2ZjYzZTkyMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgwNlQxNzUwMDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02M2E4ZTY0ZmMzOTFjYTc3OWY4ZWIyOTZlNzRlN2JhZmNjYjc1NmIyMzNkOTMwN2E2ZjgxNzA0YTY0NTU0MmNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.Ny753swM5fu9Qq_xtVB5gZ4u27tNvK_mGl7w5Mr2Fb0)
> 
> * [x]  我搜索过了 issue，但是并没有发现过与我类似的问题
> * [x]  我确认在日志中去掉了敏感信息

--- a/nonebot_bison/utils/site.py
+++ b/nonebot_bison/utils/site.py
@@ -1,5 +1,5 @@
-from abc import ABC
 from typing import Literal
+from abc import ABC, abstractmethod
 
 from httpx import AsyncClient
 
@@ -8,12 +8,16 @@ from .http import http_client
 
 
 class ClientManager(ABC):
+    @abstractmethod
     async def get_client(self, target: Target | None) -> AsyncClient: ...
 
+    @abstractmethod
     async def get_client_for_static(self) -> AsyncClient: ...
 
+    @abstractmethod
     async def get_query_name_client(self) -> AsyncClient: ...
 
+    @abstractmethod
     async def refresh_client(self): ...
 
 
@@ -26,6 +30,9 @@ class DefaultClientManager(ClientManager):
 
     async def get_query_name_client(self) -> AsyncClient:
         return http_client()
+
+    async def refresh_client(self):
+        pass
 
 
 class Site:
@@ -46,6 +53,6 @@ def anonymous_site(schedule_type: Literal["date", "interval", "cron"], schedule_
         {
             "schedule_type": schedule_type,
             "schedule_setting": schedule_setting,
-            "client_mgr": ClientManager,
+            "client_mgr": DefaultClientManager,
         },
     )

